drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -339,30 +362,41 @@ static enum ieee802154_hw_caps
 ieee802154_cc13xx_cc26xx_subg_get_capabilities(const struct device *dev)
 {
 	/* TODO: enable IEEE802154_HW_FILTER */
-	return IEEE802154_HW_FCS | IEEE802154_HW_CSMA
-	       | IEEE802154_HW_SUB_GHZ;
+	return IEEE802154_HW_FCS | IEEE802154_HW_SUB_GHZ;
 }
 
 static int ieee802154_cc13xx_cc26xx_subg_cca(const struct device *dev)
 {
 	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data = dev->data;
-	RF_Stat status;
+	RF_EventMask events;
+	int ret;
 
 	drv_data->cmd_prop_cs.status = IDLE;
 	drv_data->cmd_prop_cs.pNextOp = NULL;
 	drv_data->cmd_prop_cs.condition.rule = COND_NEVER;
 
-	status = RF_runImmediateCmd(drv_data->rf_handle,
-				    (uint32_t *)&drv_data->cmd_prop_cs);
-	if (status != RF_StatSuccess) {
-		LOG_ERR("Failed to request CCA (0x%x)", status);
+	/* TODO: Check whether RX is actually enabled. */
+	ret = ieee802154_cc13xx_cc26xx_subg_stop(dev);
+	if (ret < 0) {
 		return -EIO;
 	}
 
+	events = RF_runCmd(drv_data->rf_handle, (RF_Op *)&drv_data->cmd_prop_cs, RF_PriorityNormal,
+			   NULL, 0);
+	if (events != RF_EventLastCmdDone) {
+		LOG_DBG("Failed to request CCA: 0x%" PRIx64, events);
+		return -EIO;
+	}
+
+	/* TODO: Should we re-enable RX? Would not normally
```

Comment:
Proabably would be safe to do. I can't remember the exact details. @vaishnavachath ?